### ISSUE TITLE
Dulplicated query items when redirecting from cas login page

### ISSUE THREAD

**Issue by gordonlin** [apereo/dotnet-cas-client#19]
I encounter an issue during development using .net cas client.
Here is my original url: http://example.com/?param1=a&param2=b
After redirecting back from cas login page, the url turned out to be : http://example.com?param1=a&param2=b&param2=b

I step through the code and found that there might be wrong here:
https://github.com/Jasig/dotnet-cas-client/blob/master/DotNetCasClient/Utils/EnhancedUriBuilder.cs#L273

The querystring is decoded before splitting by '&'. It caused duplicated query items. 
In my case, the original querystring:

```
param1=a&param2=b&TARGET=http%3A%2F%2Flocalhost%3A50582%2F%3Fparam1%3Da%26param2%3Db&SAMLart=foo
```

Then after url decode:

```
param1=a&param2=b&TARGET=http://localhost:50582/?param1=a&param2=b&SAMLart=foo
```

query items after splitting by '&':
param1, param2, target, param2, samlart

In my opinion, the querystring should not be decoded before splitting by '&'.

**Comment by hokiecoder** [comment]
I believe that this issue is completely resolved by the [url-encoding-bugfix](https://github.com/apereo/dotnet-cas-client/tree/url-encoding-bugfix) branch that is a proposed fix for issue #18. The changes in that branch do exactly what **gordonlin** has asked for here. Can the two issues be merged, or perhaps this one be closed with a reference to the earlier issue?

**Comment by phantomtypist** [comment]
@TheHokieCoder So I merged your PR from #53.  Do we still need all the other changes from [url-encoding-bugfix](https://github.com/apereo/dotnet-cas-client/tree/url-encoding-bugfix) ?

**Comment by TheHokieCoder** [comment]
I don't see any additional changes between `url-encoding-bugfix` and `develop`. What other changes are you referring to? It appears that there were only 2 commits made on that branch, and the net result was included in PR #53.

**Comment by phantomtypist** [comment]
SMH I wasn't comparing the two branch... must have been looking at a different branch other than develop.

Done!  P.S. I'm going to close Scott's branch.
